blockstack_client/backend/drivers/storjstore.py

```diff
@@ -0,0 +1,468 @@
+        self.deleter = http.Client(self.user, self.passwd)
+
+    def reset(self):
+        try:
+            self.init()
+        except Exception, e:
+            log.exception(e)
+
+
+def dump2file(name, data, tmp_dir):
+    tmp_file = os.path.join(tmp_dir, name)
+    with open(tmp_file, 'w') as f:
+        f.write(data)
+    return tmp_file, tmp_dir
+
+
+def storj_encode_name(name):
+    return name.replace( "/", r"-2f" )
+
+
+def storj_get_bucket_id(user, passwd, name):
+    """
+    Ref: https://storj.github.io/core/lib_utils.js.html
+    """
+    return rip160sha256(user+name)[:24]
+
+def storj_get_file_id(user, passwd, bucket_id, name):
+    """
+    Ref: https://storj.github.io/core/lib_utils.js.html
+    """
+    return rip160sha256(bucket_id+name)[:24]
+
+def storage_init(conf, index=False, force_index=False):
+    """
+    Initialize storage driver for storj
+    """
+    global TMP_DIR, STORJ_USER, STORJ_PASSWD, STORJ_BUCKET, STORJ_CONN
+    config_path = conf['path']
+
+    if os.path.exists( config_path ):
+
+        parser = SafeConfigParser()
+
+        try:
+            parser.read(config_path)
+        except Exception, e:
+            log.exception(e)
+            return False
+
+        bucket_name = None
+        if parser.has_section('storj'):
+            if parser.has_option('storj', 'user') and parser.has_option('storj', 'passwd'):
+                STORJ_USER = parser.get('storj', 'user')
+                STORJ_PASSWD = parser.get('storj', 'passwd')
+
+            if parser.has_option("storj", "bucket"):
+                bucket_name = parser.get("storj", "bucket")
+
+    # Auth
+    if STORJ_USER is None or STORJ_PASSWD is None:
+        log.error("Config file: 'storj.user' or 'storj.passwd' is missing")
+        return False
+
+    if bucket_name is None:
+        log.error("Config file: 'storj.bucket' is missing")
+        return False
+
+
+    STORJ_BUCKET = storj_get_bucket_id(STORJ_USER, STORJ_PASSWD, bucket_name)
+    log.debug("Bucket id is %s", STORJ_BUCKET)
+    if STORJ_BUCKET is None:
+        log.error("Invalid storj bucket %s", bucket_name)
+
+    config_path = conf['path']
+    TMP_DIR = get_driver_settings_dir(config_path, ".tmp")
+    if not os.path.exists(TMP_DIR):
+        try:
+            os.makedirs(TMP_DIR)
+            os.chmod(TMP_DIR, 0700)
+        except Exception as e:
+            if DEBUG:
+                log.exception(e)
+
+    try:
+        STORJ_CONN = StorjAPI(STORJ_USER, STORJ_PASSWD, STORJ_BUCKET, TMP_DIR)
+    except Exception as e:
+        if DEBUG:
+            log.exception(e)
+        return False
+
+    return True
+
+
+def storj_put_chunk(chunk_buf, name):
+    """
+    Store a chunk to storj
+    Return URL on success
+    Return None on error
+    """
+
+
+    try:
+        name = storj_encode_name(name)
+        file_id = STORJ_CONN.write(name, chunk_buf)
+        return "storj:///blockstack/{}".format(file_id)
+    except Exception as e:
+        STORJ_CONN.reset()
```

Comment:
Ah, I see.

The driver should mask socket and connection timeouts that occur due to keeping a stale socket descriptor around.  The only times the driver is allowed to fail are (1) if the node is offline, (2) the storage system is unavailable or unable to process the request, or (3) in the case of reads, the data itself does not exist.